pkg/my/domain.go

```diff
@@ -18,40 +21,51 @@ type Domains struct {
 	domains []Domain
 }
 
-func (d *Domains) List() []Domain {
-	return d.domains
-}
-
-func (d *Domains) Save() error {
-	cnt, err := json.Marshal(d.domains)
+func GetDomains(client *node.Client, nickname string) ([]string, error) {
+	const ownedPrefix = "owned"
+	wallet, err := wallet.GetWallet(nickname)
 	if err != nil {
-		return err
+		return nil, err
 	}
-
-	err = os.WriteFile(d.file, cnt, 0o600) // u+r, u+w
+	domains := []string{}
+	domainsEntry, err := node.DatastoreEntry(client, dns.DnsRawAddress, ownedPrefix+wallet.Address)
 	if err != nil {
-		return err
+		return nil, err
 	}
-
-	return nil
+	err = json.Unmarshal(domainsEntry.CandidateValue, &domains)
+	if err != nil {
+		return nil, err
+	}
+	return domains, nil
 }
 
-func (d *Domains) Add(dom Domain) {
-	d.domains = append(d.domains, dom)
-}
+func GetOwnedWebsites(client *node.Client, domainNames []string) ([]*models.Websites, error) {
```

Comment:
What is the model linking domain and website ?

I propose to rename the function to Website, that will allow an my.Websites usage.